dev/PowerNotifications/PowerNotifications.cpp

```diff
@@ -0,0 +1,501 @@
+
+    std::mutex PowerManager::m_mutex;
+
+    int   PowerManager::m_batteryChargePercent, PowerManager::m_oldBatteryChargePercent;
+    DWORD PowerManager::m_cachedDisplayStatus;
+    DWORD PowerManager::m_cachedUserPresenceStatus;
+    DWORD PowerManager::m_cachedSystemAwayModeStatus;
+    DWORD PowerManager::m_cachedPowerCondition;
+    GUID  PowerManager::m_cachedPowerSchemePersonality;
+    ULONGLONG               PowerManager::m_cachedDischargeTime;
+    CompositeBatteryStatus *PowerManager::m_cachedCompositeBatteryStatus;
+    ::EnergySaverStatus PowerManager::m_cachedEnergySaverStatus;
+    winrt::Microsoft::ProjectReunion::BatteryStatus     PowerManager::m_batteryStatus;
+    winrt::Microsoft::ProjectReunion::BatteryStatus     PowerManager::m_oldBatteryStatus = winrt::Microsoft::ProjectReunion::BatteryStatus::Uninitialized;
+    winrt::Microsoft::ProjectReunion::PowerSupplyStatus PowerManager::m_powerSupplyStatus;
+    winrt::Microsoft::ProjectReunion::PowerSupplyStatus PowerManager::m_oldPowerSupplyStatus = winrt::Microsoft::ProjectReunion::PowerSupplyStatus::Uninitialized;
+
+
+    bool PowerManager::NotAlreadyRegisteredForEvents(PowerManager::eventType eventObj)
+    {
+        return !eventObj.get();
+    }
+
+    PowerManager::eventType PowerManager::GetEventObj(PowerFunction const& fn)
+    {
+        switch (fn) {
+        case EnergySaverStatusFn:      return PowerManager::m_energySaverStatusChangedEvent;
+        case BatteryStatusFn:          return PowerManager::m_batteryStatusChangedEvent;
+        case PowerSupplyStatusFn:      return PowerManager::m_powerSupplyStatusChangedEvent;
+        case RemainingChargePercentFn: return PowerManager::m_remainingChargePercentChangedEvent;
+        case RemainingDischargeTimeFn: return PowerManager::m_remainingDischargeTimeChangedEvent;
+        case PowerSourceStatusFn:      return PowerManager::m_powerSourceStatusChangedEvent;
+        case DisplayStatusFn:          return PowerManager::m_displayStatusChangedEvent;
+        case SystemIdleStatusFn:       return PowerManager::m_systemIdleStatusChangedEvent;
+        case PowerSchemePersonalityFn: return PowerManager::m_powerSchemePersonalityChangedEvent;
+        case UserPresenceStatusFn:     return PowerManager::m_userPresenceStatusChangedEvent;
+        case SystemAwayModeStatusFn:   return PowerManager::m_systemAwayModeStatusChangedEvent;
+        default: throw E_NOTIMPL;
+        }
+    }
+
+    EventToken PowerManager::AddCallback(PowerFunction const& fn, PowerEventHandle const& handler)
+    {
+        auto eventObj = GetEventObj(fn);
+        std::scoped_lock<std::mutex> lock(m_mutex);
+
+        //Check prior registrations
+        if (NotAlreadyRegisteredForEvents(eventObj))
+        {
+            RegisterListener(fn);
+        }
+        eventObj.get().add(handler);
+    }
+
+    void PowerManager::RegisterListener(PowerFunction const& fn)
+    {
+        switch (fn) {        
```

Comment:
Inconsistent indentation - in some places your "case" is indented, in others it's not.